api/external_notion_test.go

```diff
@@ -0,0 +1,168 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+
+	jwt "github.com/golang-jwt/jwt"
+)
+
+const (
+	notionUser           string = `{"id":"notionTestId","name":"Notion Test","picture":"http://example.com/avatar","email":"notion@example.com","verified_email":true}}`
+	notionUserWrongEmail string = `{"id":"notionTestId","name":"Notion Test","picture":"http://example.com/avatar","email":"other@example.com","verified_email":true}}`
+	notionUserNoEmail    string = `{"id":"notionTestId","name":"Notion Test","picture":"http://example.com/avatar","verified_email":true}}`
```

Comment:
haven't looked at it in detail but i'm guessing the tests aren't passing because the user info provided doesn't match the `notionUser` struct in `notion.go` which results in the data being deserialised incorrectly

Reply:
seems like most of the tests are failing because of this error message: https://github.com/supabase/gotrue/runs/4758917067?check_suite_focus=true#step:7:649 - which is why i suspect that the data is being deserialised incorrectly

Reply:
Yeah I was suspecting this to be the root cause as well. But I'm not sure how the mock server works. 
Thank you for pointing this out! 

I'll give this a try!